### DISCUSSION TITLE
Size of icon button

### DISCUSSION THREAD

**Discussion by tvojtech** [shadcn-ui/ui#6218] [Q&A] [answered]
Hi,

I have code like 
```jsx
<Button
  onClick={() => toggleDrawer()}
  aria-label="Open menu"
  className="lg:hidden"
  variant="link"
  size="icon"
  >
    <Menu /> // icon from lucide-react
</Button>
```

How can I make the icon bigger? `size` prop of the icon does not work. Setting width and height via `className` does not work on the icon or the button. The icon is still small. Can I somehow make it bigger?

**Comment by Chukwu3meka** [comment]
You can use the fontSize property

```jsx 
<Menu style={{fontSize: 40 // '40px' }} /> 
```

**Comment by HafidzNurmawanHakim** [comment]
you can use scale instead

```<Bell className="scale-125" />```
or 
```<Bell style={{ scale: 2 }} />```

**Comment by katongole-isaac** [comment, marked as answer]
I also faced the same issue. I ignore the `Button` and used `div`
```jsx
<div role="button" > <Menu size={30} /> </div> // it works fine
```

**Comment by jdgamble555** [comment]
The button base class is configured with the svg icon size, not intuitive at all. You can replace it with:

```tsx
<Button href={`/deck/${id}/${slug}`} variant="ghost" class="[&_svg]:size-6">
	<X />
</Button>
```

J

**Comment by patrice012** [comment]
When using the Button just add **[&_svg]:size-6** to the Button **className** to enable resizing just like

```
<Button
    variant="outline"
    size="icon"
    className="[&_svg]:size-6 w-1/10 h-16 px-6 border-none absolute right-4 top-1/2 transform -translate-y-1/2 hover:opacity-100 
      hover:bg-[#FFFFFF]"
  >
     <CameraIcon/>
  </Button>
```

**Comment by me-v2** [comment]
[&_svg]:size-6 is nice.

**Comment by saskia002** [comment]
![image](https://github.com/user-attachments/assets/9e465720-141c-46ed-8da9-cd7fd97e556c)


if you read through the css then this should be the correct way;

```
<Button variant="ghost" size="icon" asChild>
    <Link href="/">
      <ChevronsRight className="size-6" />
    </Link>
</Button>
```


just add `className="size-*"` to the icon and then it overrides def size value

**Reply by forrestwilkins** [reply]
This seems to be most correct and worked for me.

**Reply by wu0407** [reply]
only work on v4

**Reply by elisgrahn** [reply]
Thanks, this should be the marked answer!

**Comment by TranManhThang96** [comment]
Please add a non-existent class override, for example, **[&_svg]:size-99**, and your icon will work normally

```
<Button variant="ghost" class="[&_svg]:size-99">
	<X size={size}/>
</Button>
```

**Comment by cc-gperez** [comment]
I'd just go ahead and remove `[&_svg]:size-4 ` from the button. You have control over the Button code. Why forcing the icon size? It makes[ lucide-react sizing mechanisms](https://lucide.dev/guide/basics/sizing#resizing-with-tailwind) break.

You'll usually want to explicitly set the icons size, like `<ImageIcon className="h-6 w-6" />`

**Reply by jdgamble555** [reply]
People that update a lot don't change the code manually (keep up with bug fixes etc).  I do think this is fixed in Tailwind 4.  You could also do `[&_svg]:size-auto` to get rid of it.

**Reply by dannybunschoten** [reply]
The correct and intended way is to use the `size-*` class instead of the `h-*` and `w-*` classes.  
So in your case, you could use:

```jsx
<ImageIcon className="size-6" />
```

Check this Tailwind reference for more information: [Tailwind Docs – Setting both width and height](https://tailwindcss.com/docs/width#setting-both-width-and-height)

**Comment by subbapdm** [comment]
shadcn's [&_svg:not([class*='size-'])]:size-4 CSS rule overrides the width/height attributes, so either give classs like this [&_svg:not([class*='size-'])]:size-6 to Button component or give class size-* on lucide icons like this: className="size-6"

**Solution 1:**
`<Button variant="ghost" className="cursor-pointer text-gray-400 [&_svg:not([class*='size-'])]:size-4">
     <Bookmark />
</Button>`

**Solution 2:**
`<Button variant="ghost" className="cursor-pointer text-gray-400">
        <Bookmark className="size-6" />
</Button>`

I personally prefer giving size-* class on icon as it is short and cleanest approach.

**Comment by 81reap** [comment]
I think #7079 fixes this
